packages/next/next-server/lib/router/utils/parse-relative-url.ts

```diff
@@ -1,10 +1,19 @@
 import { getLocationOrigin } from '../../utils'
 import { searchParamsToUrlQuery } from './querystring'
 
-const DUMMY_BASE = new URL(
+let DUMMY_BASE = new URL(
   typeof window === 'undefined' ? 'http://n' : getLocationOrigin()
 )
 
+/**
+ * Refresh DUMMY_BASE for unit test
+ */
+export function refreshDummyBase() {
```

Comment:
can't you just move it into `parseRelativeUrl`?

```js
export function parseRelativeUrl(url: string, base?: string) {
  const globalBase = new URL(typeof window === 'undefined' ? 'http://n' : getLocationOrigin())
  const resolvedBase = base ? new URL(base, globalBase) : globalBase
  // ...
  if (origin !== globalBase.origin) {
  // ...
```

Reply:
We can't just move it into `parseRelativeUrl`,
because `DUMMY_BASE` assigned once when the app init,
also `DUMMY_BASE` required to be updated when under unit testing.

If move it into `parseRelativeUrl`,
the base will be dynamic,
also run the condition every time.

Reply:
`refreshDummyBase` will get included in the production bundle. As it is code that is only intended to be used for unit testing, having it in the production bundle is not an option I'm afraid. The change I propose in my comment before should make those tests work. Alternatively, now that you have electron in place, maybe these unit tests can be replaced with some integration tests?

Reply:
Agree, production bundle should not bundle `refreshDummyBase`,
Currently, here are conflict between Spectron and ChromeDriver on CI server,
The integration test pass on local but CI server.